FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -1249,22 +1705,16 @@ BaseType_t xReturn = pdTRUE;
 
 			if( ( pxDNSMessageHeader->usFlags & dnsRX_FLAGS_MASK ) == dnsEXPECTED_RX_FLAGS )
 			{
-			const uint16_t usCount = ( uint16_t ) ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY;
-			uint16_t usNumARecordsStored = 0;
-	
-				for( x = 0U; x < pxDNSMessageHeader->usAnswers; x++ )
+				const uint16_t usCount = ( uint16_t ) ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY;
+				uint16_t usNumARecordsStored = 0;
+
+				for( x = 0U; ( x < pxDNSMessageHeader->usAnswers ) && ( usNumARecordsStored < usCount ); x++ )
```

Comment:
Can we make it like the original source? It removes a MISRA violation.

Reply:
Done.